src/rust/engine/src/rule_graph.rs

```diff
@@ -679,7 +679,7 @@ impl RuleGraph {
       })
       .collect::<Vec<String>>();
     internal_rule_strs.sort();
-    write!(f, "{}\n", internal_rule_strs.join("\n"))?;
+    writeln!(f, "{}", internal_rule_strs.join("\n"))?;
     write!(f, "}}")
```

Comment:
Maybe this too, since I don't think an ending newline matters, and the consistency would be good.

Reply:
Done :)